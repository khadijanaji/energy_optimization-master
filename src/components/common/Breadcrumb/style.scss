ol {
    list-style: none;
    display: flex;
    
    margin-block-start: 0;
    margin-block-end: 0;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    padding-inline-start: 0;
    
    
    
    
   
}

.breadcrumb {
  &__container {
    //margin-top: -1px;
    box-shadow: 0 4px 0 rgba(0, 0, 0, 0.12), 0 4px -1px 0 rgba(0, 0, 0, 0.24);
    padding: 0 0 12px 24px;
    width: calc(100% - 24px);
    //z-index: 98;
    //position: sticky;
    overflow: scroll;
    background-color: #263237;
  }
  &__previous {
    color: #FFF5DEB3;
  }
  &__current {
    color: #8092A5;
  }
  &__link{
    color: inherit;
  }
  &__separator{
    padding: 2px 8px 2px 8px;
    color: #8092A5;
    font-weight: bolder;
  }
}