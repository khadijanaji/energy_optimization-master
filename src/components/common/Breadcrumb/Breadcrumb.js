/* eslint-disable react/prop-types */
/* eslint-disable react/no-multi-comp */
// @flow
import React from "react";
import "./style.scss";

const BreadcrumbItem = ({children, ...props}) => (
  <li
    className="breadcrumb-item"
    {...props}
  >
    {children}
  </li>
);


const toBreadcrumbItem = (child, index) => (
  <BreadcrumbItem key={`breadcrumb_item${index}`}>{child}</BreadcrumbItem>
);

const withSeparator = (lastIndex) => (acc, child, index) => {
  const notLast = index < lastIndex;
  if (notLast) {
    acc.push([
      <span
        className="breadcrumb__previous"
        key={`element-${index}`}
      >{child}</span>,
      <span
        className="breadcrumb__separator"
        key={`separator-${index}`}
      >{" > "}</span>
    ]);
  } else {
    acc.push(<span
      className="breadcrumb__current"
      key={`element-${index}`}
    >{child}</span>);
  }
  return acc;
};

const Breadcrumb = ({...props}: any) => {
  let {children} = props;
  children = React.Children.toArray(children);

  const totalItems = children.length;
  const lastIndex = totalItems - 1;

  children = children.map(toBreadcrumbItem).reduce(withSeparator(lastIndex), []);

  return <div className="breadcrumb__container"><ol>{children}</ol></div>;
};

export default Breadcrumb;
