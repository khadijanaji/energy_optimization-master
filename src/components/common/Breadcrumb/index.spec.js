/* globals describe it expect */

import {shallow} from "enzyme";
import React from "react";

import Breadcrumb from "./";

describe("Component", () => {
  describe("Breadcrumb Render", () => {
    it("Without Props: Should match snapshot", () => {
      const wrapper = shallow(<Breadcrumb />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
