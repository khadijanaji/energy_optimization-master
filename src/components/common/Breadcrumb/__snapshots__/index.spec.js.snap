// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Breadcrumb Render Without Props: Should match snapshot 1`] = `
<div
  className="breadcrumb__container"
>
  <ol />
</div>
`;
